File: Unit 2/IN300_WilliamAlecAkin_Unit2_WHILE_Loop.r
# William Alec Akin
# Purdue University Global
# IN300 - Programming for Data Analaysis (Python, R, Java)
# Dr. Robert Kayl
# Unit 2 Assignment - WHILE Loop - R Source Code
# May 8th, 2022

# List of book categories for menu:
# 1. Science Fiction
# 2. Computers and Technology
# 3. Cooking
# 4. Business
# 5. Comics
# 6. Exit

#Psudo-code for R:
# 0. Print a welcome message to the user
# 1. Use while loop to iterate through the menu options and print to stdout
# 2. Ask the user to choose an integer between 1 and 6
# 3. Based on the user's choice, output the categor and a complimentary message
# 4. Return the user to the menu to chose another category or exit

# Print a welcome message to the user
print("Welcome to the book category menu!")

# Use while loop to iterate through the menu options and print to stdout
while (TRUE) {
    # Ask the user to choose an integer between 1 and 6
    print("Please choose a category:")
    print("1. Science Fiction")
    print("2. Computers and Technology")
    print("3. Cooking")
    print("4. Business")
    print("5. Comics")
    print("6. Exit")

    # Ask for user's input
    user_input <- readline(prompt = "Please enter a number between 1 and 6: ")

    # Convert the user's input to an integer
    user_input <- as.numeric(user_input)

    # Based on user's choice, output the category and a complimentary message
    # Final else statement performs error validation
    if (user_input == 1) {
        print("You have chosen Science Fiction.")
        print("Science Fiction is a genre of fiction in which imaginative fiction is set in a fictional universe.")
    } else if (user_input == 2) {
        print("You have chosen Computers and Technology.")
        print("Computers and Technology is about the use of computers and technology to solve problems.")
    } else if (user_input == 3) {
        print("You have chosen Cooking.")
        print("Cooking is the art of preparing food.")
    } else if (user_input == 4) {
        print("You have chosen Business.")
        print("Business is the business of doing business.")
    } else if (user_input == 5) {
        print("You have chosen Comics.")
        print("Comics is a genre of literature that portrays images and stories in comic strips.")
    } else if(user_input == 6) {
        print("You have chosen to exit the program.")
        print("Thank you for using the book category menu!")
        break
    } else {
        print("Invalid input. Please enter a number between 1 and 6.")
    }
}

File: Unit 2/IN300_WilliamAlecAkin_Unit2_WHILE_Loop.java
/* William Alec Akin
Purdue University Global
IN300 - Programming for Data Analaysis (Python, R, Java)
Dr. Robert Kayl
Unit 2 Assignment - WHILE Loop - Java Source Code
May 8th, 2022
*/

/* List of book categories for menu:
1. Science Fiction
2. Computers and Technology
3. Cooking
4. Business
5. Comics
6. Exit
*/

/*
Psudo-code:
1. Use while loop to iterate through the menu options and print to stdout
2. Ask the user to choose an integer between 1 and 6
3. Based on the user's choice, output the categor and a complimentary message
4. Return the user to the menu to chose another category or exit
*/

//Import necessary libraries
import java.util.Scanner;

//Create main Java class where all the magic happens
public class IN300_WilliamAlecAkin_Unit2_WHILE_Loop {
    public static void main(String[] args) 
    {
        //Create scanner object to read in user input
        Scanner scanner = new Scanner(System.in);

        // Define ANSI color codes equal to variables to make them easier to use
        //Create variables which allow use of simple ANSI_COLOR references instead of the codes themselves
        String ANSI_RESET = "\u001B[0m";
        String ANSI_BLUE = "\u001B[34m";
        String ANSI_GREEN = "\u001B[32m";
        String ANSI_YELLOW = "\u001b[33m";
        String ANSI_MAGENTA = "\u001b[35m";
        String ANSI_CYAN = "\u001b[36m";
        String ANSI_RED = "\u001b[31m";

        //Define variables to create a counter for the while loop
        //int counter = 0;
        int choice = 0;

        //Print welcome statement to stdout
        System.out.println(ANSI_GREEN + "Welcome to the book category menu!" + ANSI_RESET);

        //Use a while loop to iterate through the menu options and print to stdout and iterate the counter by 1
        while (true){
            System.out.println("1. Science Fiction");
            System.out.println("2. Computers and Technology");
            System.out.println("3. Cooking");
            System.out.println("4. Business");
            System.out.println("5. Comics");
            System.out.println("6. Exit");
            //counter++;
            
            //Prompt the user to enter a number between 1 and 6 and assign it to the choice variable
            System.out.println(ANSI_BLUE + "Please enter a number between 1 and 6: " + ANSI_RESET);
            choice = scanner.nextInt();
            System.out.println();

            //Use IF/ELSE IF/ELSE statements to print the appropriate category and message based on the user's choice
            if(choice == 1)
            {
                System.out.println(ANSI_MAGENTA + "You chose Science Fiction!");
                System.out.println("You are a fan of science fiction!" + ANSI_RESET);
                System.out.println();
                System.out.println();
            }
            else if(choice == 2)
            {
                System.out.println(ANSI_CYAN + "You chose Computers and Technology!");
                System.out.println("Computers and Technology is the study of computers and their applications." + ANSI_RESET);
                System.out.println();
                System.out.println();
            }
            else if (choice == 3)
            {
                System.out.println(ANSI_YELLOW + "You chose Cooking!");
                System.out.println("Cooking is the art of preparing food." + ANSI_RESET);
                System.out.println();
                System.out.println();
            }
            else if (choice == 4)
            {
                System.out.println(ANSI_RED + "You chose Business!");
                System.out.println("Business is the study of the business world." + ANSI_RESET);
                System.out.println();
                System.out.println();
            }
            else if (choice == 5)
            {
                System.out.println(ANSI_BLUE + "You chose Comics!");
                System.out.println("Comics is the study of comic books." + ANSI_RESET);
                System.out.println();
                System.out.println();
            }
            else if (choice == 6)
            {
                System.out.println(ANSI_GREEN + "You chose to exit the program!");
                System.out.println("Thank you for using the program!" + ANSI_RESET);
                System.out.println();
                System.out.println();
                break;
            
        }
    }
}
}
